p2p/src/simulated/network.rs

```diff
@@ -251,12 +258,16 @@ impl<E: RNetwork + Spawner + Rng + Clock + Metrics, P: Array> Network<E, P> {
                 continue;
             }
 
+            // Determine if the sender or recipient has blocked the other
+            let o_r = (origin.clone(), recipient.clone());
+            let r_o = (recipient.clone(), origin.clone());
+            if self.blocks.contains(&o_r) || self.blocks.contains(&r_o) {
```

Comment:
I'm surprised there isn't a way to do this without `clone()` but 🤷

Reply:
Yeah only if you iterate through the map manually I think. Could potentially avoid two checks by ordering (canonicalizing) the tuple first

Reply:
I think this is fine as-is for now 👍